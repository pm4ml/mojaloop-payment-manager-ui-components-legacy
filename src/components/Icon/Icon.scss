@import '../../assets/styles/vars/colors.scss';

.el-icon {
  display: flex;
  flex: 0 0 auto;
}

.el-icon--spin {
  animation: spinIcon 1s linear infinite;
}

@keyframes spinIcon {
  0% {
    transform: rotate(0deg);
  }

  100% {
    transform: rotate(360deg);
  }
}
