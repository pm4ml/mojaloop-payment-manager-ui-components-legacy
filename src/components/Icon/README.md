# Icon

Renders a SVG icon by using the xLinkHref reference. The SVG should already be imported in order to be used.

### Proptypes

|       Name      |       Type       |              Description               | Optional |                  Values                  |   Default   |
|-----------------|------------------|----------------------------------------|----------|------------------------------------------|-------------|
| classname       | `string`         | defines the HTML classname             | x        |                                          | `undefined` |
| id              | `string`         | defines the HTML id                    | x        |                                          | `undefined` |
| style           | `object`         | defines the HTML style                 | x        |                                          | `undefined` |
| name            | `string`         | defines the icon name _xLinkHref_      |          |                                          | `undefined` |
| fill            | `string`         | defines the fill color                 |          |                                          | `undefined` |
| stroke          | `string`         | defines the stroke color               |          |                                          | `undefined` |
| spin            | `bool`           | defines whether the icon should rotate |          |                                          | `false`     |
| size            | `number`         | defines the icon height and width      |          |                                          | `20`        |
| tooltip         | `node`, `string` | defines the tooltip component or label |          |                                          | `undefined` |
| tooltipDelay    | `number`         | defines the delay in ms                |          |                                          | `undefined` |
| tooltipKind     | `string`         | defines the tooltip kind               |          | _regular, error, info, warning, neutral_ | `undefined` |
| tooltipPosition | `string`         | defines the tooltip position           |          | _top, bottom, left, right_               | `undefined` |
