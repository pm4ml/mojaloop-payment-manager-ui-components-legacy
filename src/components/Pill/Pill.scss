@import '../../assets/styles/vars/colors.scss';
@import '../../assets/styles/vars/shadow.scss';

.el-pill {
  border-radius: 20px;
  border: 2px solid;
  min-height: 24px;
  padding: 3px 7px;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  cursor: initial;
}

.el-pill--active {
  background: #eee;
  color: #999;
}

.el-pill--default {
  fill: #333;
  color: #333;
  border-color: #333;
  background-color: #eee;
}
.el-pill--primary {
  fill: $color__main-primary;
  color: $color__main-primary;
  border-color: $color__main-primary;
  background-color: $shadow-color--primary-2;
}
.el-pill--secondary {
  fill: $color__main-secondary;
  color: $color__main-secondary;
  border-color: $color__main-secondary;
  background-color: $shadow-color--secondary-2;
}
.el-pill--tertiary {
  fill: $color__main-tertiary;
  color: $color__main-tertiary;
  border-color: $color__main-tertiary;
  background-color: $shadow-color--tertiary-2;
}
.el-pill--success {
  fill: $color__main-success;
  color: $color__main-success;
  border-color: $color__main-success;
  background-color: $shadow-color--success-2;
}
.el-pill--danger {
  fill: $color__main-danger;
  color: $color__main-danger;
  border-color: $color__main-danger;
  background-color: $shadow-color--danger-2;
}
.el-pill--warning {
  fill: $color__main-warning;
  color: $color__main-warning;
  border-color: $color__main-warning;
  background-color: $shadow-color--warning-2;
}
.el-pill--dark {
  fill: #000;
  color: #000;
  border-color: #000;
  background-color: $shadow-color--dark-2;
}
.el-pill--light {
  fill: $color__white;
  color: $color__white;
  border-color: $color__white;
  background-color: shadow-color-0light;
}
.el-pill--inactive {
  border-color: transparent;
}

.el-pill--active {
}

.el-pill__icon-box {
  flex: 0 0 auto;
  display: flex;
  align-items: center;
  justify-content: center;
  margin: 0px 2px;
}
.el-pill__label {
  display: inline-flex;
  align-items: center;
  justify-content: center;
  font-size: 12px;
  font-weight: bold;
  margin: 0px 2px;
  cursor: initial;
}
.el-pill__icon {
}
