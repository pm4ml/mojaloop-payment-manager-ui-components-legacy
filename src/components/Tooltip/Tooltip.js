import './Tooltip.scss';

import delay from 'lodash/delay';
import PropTypes from 'prop-types';
import React, { PureComponent } from 'react';

import * as utils from '../../utils/common';
import uuid from '../../utils/uuid';
import TooltipViewer from './TooltipViewer';

/* eslint-disable no-mixed-operators */

class Tooltip extends PureComponent {
  static getInnerOverflow(element) {
    return element.scrollWidth > element.offsetWidth || element.scrollHeight > element.offsetHeight;
  }
  static visibleAfterScroll(element, parents) {
    const percentX = 100;
    const percentY = 100;
    const tolerance = 0.05;
    const rect = element.getBoundingClientRect();
    return parents.every(parent => {
      const prect = parent.getBoundingClientRect();
      const visiblePixelX = Math.min(rect.right, prect.right) - Math.max(rect.left, prect.left);
      const visiblePixelY = Math.min(rect.bottom, prect.bottom) - Math.max(rect.top, prect.top);

      const visiblePercentageX = (visiblePixelX / rect.width) * 100;
      const visiblePercentageY = (visiblePixelY / rect.height) * 100;
      return visiblePercentageX + tolerance > percentX && visiblePercentageY + tolerance > percentY;
    });
  }
  constructor(props) {
    super(props);
    this._scrollNodes = [];
    this._id = uuid();
    this.mountTooltip = this.mountTooltip.bind(this);
    this.unmountTooltip = this.unmountTooltip.bind(this);
    this.showTooltip = this.showTooltip.bind(this);
    this.hideTooltip = this.hideTooltip.bind(this);
    this.delayShowTooltip = this.delayShowTooltip.bind(this);
    this.delayHideTooltip = this.delayHideTooltip.bind(this);
    this.hideTooltipBeforeScroll = this.hideTooltipBeforeScroll.bind(this);
    this.showForcedTooltipAfterScroll = this.showForcedTooltipAfterScroll.bind(this);
    this.state = { show: false };
  }
  componentDidMount() {
    this._mounted = true;
    if (this.props.forceVisibility !== undefined) {
      if (this.props.forceVisibility === true) {
        this.delayShowTooltip();
      } else {
        this.delayHideTooltip();
      }
    } else {
      this.detectTooltipRequired();
    }
  }
  componentDidUpdate(prevProps, prevState) {
    if (this._scrolling) {
      return;
    }
    if (this.props.forceVisibility !== undefined) {
      if (this.props.forceVisibility === true) {
        this.delayShowTooltip();
      } else {
        this.hideTooltip();
      }
    } else {
      if (prevState.show === false && this.state.show === true) {
        return;
      }
      this.detectTooltipRequired();
    }
  }
  componentWillUnmount() {
    this._mounted = false;
    this.unmountTooltip();
  }
  delayShowTooltip() {
    this._isHoveringTooltip = this.props.forceVisibility !== true;
    delay(() => this.showTooltip(), this.props.delay);
  }
  delayHideTooltip(delayTime = this.props.delay) {
    this._isHoveringTooltip = false;
    if (this.props.forceVisibility === true) {
      return;
    }
    delay(this.hideTooltip, delayTime);
  }
  detectTooltipRequired() {
    const { content, label } = this.props;
    const hasOverflow = Tooltip.getInnerOverflow(this.container);

    if (hasOverflow || label || content) {
      this.mountTooltip();
    } else {
      this.unmountTooltip();
    }
  }
  mountTooltip() {
    if (this.props.showOnHover !== false) {
      this.container.addEventListener('mouseenter', this.delayShowTooltip);
      this.container.addEventListener('mouseleave', this.delayHideTooltip);
    }
    this.container.classList.remove('el-tooltip--inactive');
  }
  unmountTooltip() {
    clearTimeout(this.tooltipTimeout);
    this.hideTooltip();
    if (this.props.showOnHover !== false) {
      this.container.removeEventListener('mouseenter', this.delayShowTooltip);
      this.container.removeEventListener('mouseleave', this.delayHideTooltip);
    }
    this.container.classList.add('el-tooltip--inactive');
  }
  showTooltip() {
    if (!this._mounted || !this._id || !this.container) {
      // stop if not mounted, not existing, no container available
      return;
    }
    if (this._isHoveringTooltip === false && this.props.forceVisibility !== true) {
      // do show tooltip afterDealy if not still hovered and not forced
      return;
    }
    if (!this._scrollNodes.length) {
      // listen for scroll in containers
      this._scrollNodes = utils.getScrollParents(this.container);
      this._scrollNodes.forEach(node => {
        node.addEventListener('scroll', this.hideTooltipBeforeScroll);
      });
    }

    if (!Tooltip.visibleAfterScroll(this.container, this._scrollNodes)) {
      return;
    }
    this.setState({ show: true });
  }
  hideTooltip() {
    if (this._mounted === false) {
      return;
    }
    if (this._isHoveringTooltip === true && this.props.forceVisibility === false) {
      return;
    }
    if (!this.state.show) {
      return;
    }
    if (this._scrollNodes.length) {
      this._scrollNodes.forEach(node => {
        node.removeEventListener('scroll', this.hideTooltipBeforeScroll);
      });
      this._scrollNodes = [];
    }
    this.setState({ show: false });
  }
  hideTooltipBeforeScroll() {
    this._scrolling = true;
    this.hideTooltip();
    clearTimeout(this.tooltipTimeout);
    this.tooltipTimeout = setTimeout(this.showForcedTooltipAfterScroll, 500);
  }
  showForcedTooltipAfterScroll() {
    this._scrolling = false;
    this.showTooltip();
  }

  render() {
    const {
      className,
      style,
      children,
      content,
      label,
      position,
      align,
      kind,
      custom,
    } = this.props;
    const tooltipClassName = utils.composeClassNames([
      'el-tooltip',
      custom && 'el-tooltip--custom',
      className,
    ]);

    let viewer = null;
    if (this.state.show) {
      viewer = (
        <TooltipViewer
          parentId={this._id}
          content={content}
          label={label}
          position={position}
          align={align}
          kind={kind}
          custom={custom}
        >
          {children}
        </TooltipViewer>
      );
    }

    return (
      <div
        className={tooltipClassName}
        style={style}
        ref={container => {
          this.container = container;
        }}
        id={this._id}
      >
        {children}
        {viewer}
      </div>
    );
  }
}

Tooltip.propTypes = {
  delay: PropTypes.number,
  forceVisibility: PropTypes.bool,
  showOnHover: PropTypes.bool,
  content: PropTypes.node,
  children: PropTypes.node,
  className: PropTypes.string,
  style: PropTypes.shape(),
  label: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.arrayOf(PropTypes.string),
    PropTypes.node,
  ]),
  position: PropTypes.oneOf(['top', 'bottom', 'left', 'right']),
  align: PropTypes.oneOf(['start', 'center', 'end']),
  kind: PropTypes.oneOf(['regular', 'error', 'info', 'warning', 'neutral']),
  custom: PropTypes.bool,
};
Tooltip.defaultProps = {
  delay: 200,
  forceVisibility: undefined,
  showOnHover: undefined,
  content: undefined,
  children: null,
  className: undefined,
  style: {},
  label: undefined,
  position: undefined,
  align: undefined,
  kind: 'regular',
  custom: false,
};

export default Tooltip;
