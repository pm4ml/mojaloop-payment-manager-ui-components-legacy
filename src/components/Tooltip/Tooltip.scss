@import '../../assets/styles/vars/colors.scss';
@import '../../assets/styles/vars/fonts.scss';

.el-tooltip {
  max-width: 100%;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  flex: 0 0 auto;
}
.el-tooltip--custom {
  overflow: unset;
  text-overflow: unset;
  white-space: unset;
}
.el-tooltip--inactive {
  max-width: none;
  overflow: visible;
  text-overflow: unset;
  white-space: unset;
}
.el-tooltip__content {
}

.el-tooltip__child {
  position: relative;
}

.el-tooltip__child--custom {
  overflow: visible;
  color: #333;
  display: table;
}

.el-tooltip__viewer {
  display: block;
  position: absolute;
  z-index: 1000000;
  opacity: 0;
  transition: opacity 0.1s ease-in-out;
  max-width: calc(100% - 20px);
  display: table;
  font-family: $font__primary, $font__primary-fallback;
  letter-spacing: $font__primary-spacing;
}

.el-tooltip__viewer--break {
}
.el-tooltip__viewer--default {
  padding: 7px;
  border-radius: 4px;
  font-size: 12px;
  line-height: 14px;
}
.el-tooltip__viewer--regular {
  background: #000;
  color: #fff;
  fill: #fff;
}

.el-tooltip__viewer--error {
  color: #fff;
  fill: #fff;
  background: $color__main-danger;
}
.el-tooltip__viewer--info {
  color: #fff;
  fill: #fff;
  background: $color__main-primary;
}
.el-tooltip__viewer--warning {
  color: #fff;
  fill: #fff;
  background: $color__main-warning;
}

.el-tooltip__viewer--fade-in-top {
  animation-name: fade-in-top;
}
.el-tooltip__viewer--fade-in-left {
  animation-name: fade-in-left;
}
.el-tooltip__viewer--fade-in-bottom {
  animation-name: fade-in-bottom;
}
.el-tooltip__viewer--fade-in-right {
  animation-name: fade-in-right;
}
.el-tooltip__viewer--fade-in {
  opacity: 1;
  transform: translate(0px, 0px);
  animation-duration: 0.1s;
  animation-iteration-count: 1;
  animation-direction: forwards;
  animation-timing-function: ease-in-out;
}

.el-tooltip__handle {
  position: absolute;
  width: 10px;
  height: 10px;
  overflow: hidden;
  background: inherit;
  border-bottom-left-radius: 2px;
  transform: rotateZ(45deg);
  top: 2.1px;
  left: 2.1px;
}
.el-tooltip__handle-wrapper {
  position: absolute;
  width: 7.1px;
  height: 14.2px;
  overflow: hidden;
  background: none;
}
.el-tooltip__handle--default {
  background: #000;
  color: #fff;
  fill: #fff;
  box-shadow: none;
  border: none;
}
.el-tooltip__handle--error {
  background: $color__main-danger;
}
.el-tooltip__handle--info {
  background: $color__main-primary;
}
.el-tooltip__handle--warning {
  background: $color__main-warning;
}
.el-tooltip__handle--neutral {
  background: $color__white;
  box-shadow: 0px 0px 2px $color__alpha-2;
}
.el-tooltip__handle-wrapper--top-start {
  top: 100%;
  left: 14.5px;
  margin-top: -5px;
  margin-left: -4.5px;
  transform: rotateZ(270deg);
}
.el-tooltip__handle-wrapper--top-center {
  top: 100%;
  left: 50%;
  margin-top: -5px;
  margin-left: -4.5px;
  transform: rotateZ(270deg);
}
.el-tooltip__handle-wrapper--top-end {
  top: 100%;
  right: 10px;
  margin-top: -5px;
  margin-left: -4.5px;
  transform: rotateZ(270deg);
}

.el-tooltip__handle-wrapper--left-start {
  top: 14.5px;
  left: 100%;
  margin-top: -7px;
  margin-left: -1px;
  transform: rotateZ(180deg);
}
.el-tooltip__handle-wrapper--left-center {
  top: 50%;
  left: 100%;
  margin-top: -7px;
  margin-left: -1px;
  transform: rotateZ(180deg);
}
.el-tooltip__handle-wrapper--left-end {
  bottom: 9px;
  left: 100%;
  margin-top: -7px;
  margin-left: -1px;
  transform: rotateZ(180deg);
}

.el-tooltip__handle-wrapper--right-start {
  top: 14.5px;
  left: 0%;
  margin-top: -7px;
  margin-left: -6px;
  transform: rotateZ(0deg);
}

.el-tooltip__handle-wrapper--right-center {
  top: 50%;
  left: 0%;
  margin-top: -7px;
  margin-left: -6px;
  transform: rotateZ(0deg);
}
.el-tooltip__handle-wrapper--right-end {
  bottom: 9px;
  left: 0%;
  margin-top: -7px;
  margin-left: -6px;
  transform: rotateZ(0deg);
}
.el-tooltip__handle-wrapper--bottom-start {
  top: 0%;
  left: 14.5px;
  margin-top: -9px;
  margin-left: -4.5px;
  transform: rotateZ(90deg);
}
.el-tooltip__handle-wrapper--bottom-center {
  top: 0%;
  left: 50%;
  margin-top: -9px;
  margin-left: -4.5px;
  transform: rotateZ(90deg);
}
.el-tooltip__handle-wrapper--bottom-end {
  top: 0%;
  right: 10px;
  margin-top: -9px;
  margin-left: -4.5px;
  transform: rotateZ(90deg);
}

@keyframes fade-in-top {
  from {
    transform: translate(0px, -3px);
  }
  to {
    transform: translate(0px, 0px);
  }
}
@keyframes fade-in-bottom {
  from {
    transform: translate(0px, 3px);
  }
  to {
    transform: translate(0px, 0px);
  }
}
@keyframes fade-in-left {
  from {
    transform: translate(-3px, 0px);
  }
  to {
    transform: translate(0px, 0px);
  }
}
@keyframes fade-in-right {
  from {
    transform: translate(3px, 0px);
  }
  to {
    transform: translate(0px, 0px);
  }
}
