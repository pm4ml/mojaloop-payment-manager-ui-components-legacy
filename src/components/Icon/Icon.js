import './Icon.scss';

import PropTypes from 'prop-types';
import React, { PureComponent } from 'react';

import * as utils from '../../utils/common';
import Tooltip from '../Tooltip';

const iconSizes = {
  xs: 10,
  s: 12,
  m: 14,
  l: 16,
};

class Icon extends PureComponent {
  constructor(props) {
    super(props);
    this.onClick = this.onClick.bind(this);
  }

  onClick(e) {
    if (this.props.onClick) {
      this.props.onClick(e);
    }
  }
  render() {
    const {
      id,
      className,
      style,
      size,
      name,
      fill,
      stroke,
      spin,
      tooltip,
      tooltipKind,
      tooltipDelay,
      tooltipPosition,
    } = this.props;

    const svgStyle = {
      height: `${size}px`,
      width: `${size}px`,
      fill,
      stroke,
      ...style,
    };
    const componentClassName = utils.composeClassNames([
      'el-icon',
      spin && 'el-icon--spin',
      className,
    ]);
    const svg = (
      <svg id={id} style={svgStyle} onClick={this.onClick} className={componentClassName}>
        <use xlinkHref={`#${name}`} />
      </svg>
    );
    if (tooltip) {
      return (
        <Tooltip label={tooltip} position={tooltipPosition} delay={tooltipDelay} kind={tooltipKind}>
          {svg}
        </Tooltip>
      );
    }
    return svg;
  }
}

Icon.propTypes = {
  id: PropTypes.string,
  style: PropTypes.shape(),
  className: PropTypes.string,
  size: PropTypes.number,
  name: PropTypes.string,
  fill: PropTypes.string,
  stroke: PropTypes.string,
  spin: PropTypes.bool,
  onClick: PropTypes.func,
  tooltipKind: PropTypes.oneOf(['regular', 'error', 'info', 'warning', 'neutral']),
  tooltipDelay: PropTypes.number,
  tooltip: PropTypes.oneOfType([PropTypes.node, PropTypes.string]),
  tooltipPosition: PropTypes.oneOf(['top', 'bottom', 'left', 'right']),
};
Icon.defaultProps = {
  id: undefined,
  style: undefined,
  className: undefined,
  size: 20,
  name: undefined,
  fill: undefined,
  stroke: undefined,
  spin: false,
  onClick: undefined,
  tooltipKind: undefined,
  tooltipDelay: undefined,
  tooltip: undefined,
  tooltipPosition: undefined,
};

export default Icon;
export { iconSizes };
