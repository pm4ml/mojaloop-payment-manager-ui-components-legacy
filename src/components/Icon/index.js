import Icon, { iconSizes } from './Icon';

export default Icon;
export { iconSizes };
