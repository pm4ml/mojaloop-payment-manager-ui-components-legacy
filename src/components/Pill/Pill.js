import './Pill.scss';

import PropTypes from 'prop-types';
import React from 'react';

import { composeClassNames } from '../../utils/common';
import Icon from '../Icon';
import Tooltip from '../Tooltip';

const Pill = ({ id, className, kind = 'default', icon, fill, label, active, tooltip }) => {
  const pillClassName = composeClassNames([
    'el',
    'el-pill',
    `el-pill--${kind}`,
    active ? `el-pill--active` : `el-pill--inactive`,
    className,
  ]);

  let pill = (
    <div className={pillClassName} id={id}>
      {icon && (
        <div className="el-pill__icon-box">
          <Icon name={icon} size={16} className="el-pill__icon" fill={fill} />
        </div>
      )}
      {label && <span className="el-pill__label">{label}</span>}
    </div>
  );

  if (tooltip) {
    pill = <Tooltip label={tooltip}>{pill}</Tooltip>;
  }
  return pill;
};

export default Pill;

Pill.propTypes = {
  id: PropTypes.string,
  className: PropTypes.string,
  kind: PropTypes.oneOf([
    'default',
    'primary',
    'secondary',
    'tertiary',
    'success',
    'danger',
    'warning',
    'dark',
    'light',
  ]),
  icon: PropTypes.string,
  fill: PropTypes.string,
  label: PropTypes.string,
  active: PropTypes.bool,
  tooltip: PropTypes.string,
};
