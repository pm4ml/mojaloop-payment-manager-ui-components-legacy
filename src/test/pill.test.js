import { mount, shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';

import Pill from '../components/Pill';
import Icon from '../components/Icon';
import Tooltip from '../components/Tooltip';

// Shallow render, no sub-components ( still available children html)
it('renders the component', () => {
  const wrapper = shallow(<Pill icon="deploy-small" />);
  expect(wrapper.find('div.el-pill').exists()).toBe(true);
  expect(wrapper.find(Icon).exists()).toBe(true);
});

it('renders the label', () => {
  const wrapper = shallow(<Pill icon="deploy-small" label="Hello!" />);
  expect(wrapper.find('.el-pill__label')).toHaveLength(1);
  expect(wrapper.find('.el-pill__label').text()).toBe('Hello!');
});

it('renders the tooltip', () => {
  const wrapper = shallow(<Pill icon="deploy-small" tooltip="I am the tooltip" />);
  expect(wrapper.find(Tooltip)).toHaveLength(1);
});

it('renders the active state', () => {
  const wrapper = shallow(<Pill icon="deploy-small" active />);
  expect(wrapper.find('div.el-pill').hasClass('el-pill--active')).toBe(true);
});

it('renders the icon', () => {
  const wrapper = shallow(<Pill icon="deploy" />);
  expect(wrapper.find('.el-pill__icon')).toHaveLength(1);
  expect(wrapper.find(Icon)).toHaveLength(1);
});

it('applies the prop fill', () => {
  const wrapper = shallow(<Pill icon="deploy" fill="#fff" />);
  expect(wrapper.find(Icon).prop('fill')).toBe('#fff');
});

it('applies the prop className', () => {
  const wrapper = shallow(<Pill icon="deploy" className="test" />);
  expect(wrapper.find('.test')).toHaveLength(1);
});

it('renders the prop id', () => {
  const wrapper = shallow(<Pill icon="deploy" id="testPillId" />);
  expect(wrapper.find('#testPillId')).toHaveLength(1);
  expect(wrapper.find('div.el-pill').prop('id')).toBe('testPillId');
});

it('renders the default correct kind', () => {
  const wrapper = shallow(<Pill icon="deploy" />);
  expect(wrapper.find('div.el-pill').hasClass('el-pill--default')).toBe(true);
});

it('renders all the kinds', () => {
  ['default', 'primary', 'secondary', 'tertiary', 'danger', 'warning', 'dark', 'light'].forEach(
    kind => {
      const wrapper = shallow(<Pill icon="deploy" kind={kind} />);
      expect(wrapper.find('div.el-pill').hasClass(`el-pill--${kind}`)).toBe(true);
    },
  );
});

// Mount render, include sub-components

it('renders the inner icon component', () => {
  const wrapper = mount(<Pill icon="deploy" />);
  expect(wrapper.find('.el-icon')).toHaveLength(1);
});

// Snapshot testing

it('renders the button correctly when multiple props are set', () => {
  const wrapper = shallow(<Pill icon="deploy-small" kind="secondary" disabled onClick={jest.fn} />);
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});
