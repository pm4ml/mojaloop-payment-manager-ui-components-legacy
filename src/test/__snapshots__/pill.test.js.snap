// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the button correctly when multiple props are set 1`] = `
<div
  className="el el-pill el-pill--secondary el-pill--inactive"
>
  <div
    className="el-pill__icon-box"
  >
    <Icon
      className="el-pill__icon"
      name="deploy-small"
      size={16}
      spin={false}
    />
  </div>
</div>
`;
